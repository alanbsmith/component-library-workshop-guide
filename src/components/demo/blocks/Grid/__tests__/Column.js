import React from "react";
import Column from "../Column";
import "jest-styled-components";

import { renderWithTheme } from "__tests__/helpers";

describe("Grid.Column Element", () => {
  it("renders correctly", () => {
    const tree = renderWithTheme(<Column />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("applies the correct style with the `flex` prop", () => {
    const tree = renderWithTheme(<Column flex="1" />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule("flex", "1");
  });

  it("applies the correct style with the `position` modifier", () => {
    const tree = renderWithTheme(<Column modifiers={['positionRelative']} />).toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule("position", "relative");
  });
});
