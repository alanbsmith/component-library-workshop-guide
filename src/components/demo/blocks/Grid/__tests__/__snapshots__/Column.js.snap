// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid.Column Element applies the correct style with the \`flex\` prop 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0.25rem;
}

<div
  className="c0"
/>
`;

exports[`Grid.Column Element applies the correct style with the \`position\` modifier 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: initial;
  -ms-flex: initial;
  flex: initial;
  padding: 0.25rem;
  position: relative;
}

<div
  className="c0"
/>
`;

exports[`Grid.Column Element renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: initial;
  -ms-flex: initial;
  flex: initial;
  padding: 0.25rem;
}

<div
  className="c0"
/>
`;
